pyrit/memory/azure_sql_memory.py

```
@@ -349,6 +349,34 @@ def query_entries(
                 logger.exception(f"Error fetching data from table {model.__tablename__}: {e}")
                 return []
 
+    def update_entries(self, *, entries: MutableSequence[Base], update_fields: dict) -> bool:  # type: ignore
+        """
+        Updates the given entries with the specified field values.
+
+        Args:
+            entries (list[Base]): A list of SQLAlchemy model instances to be updated.
+            update_fields (dict): A dictionary of field names and their new values.
+        """
+        with closing(self.get_session()) as session:
+            try:
+                for entry in entries:
+                    # Ensure the entry is attached to the session. If it's detached, merge it.
+                    entry_in_session = session.merge(entry)
+                    for field, value in update_fields.items():
+                        if hasattr(entry_in_session, field):
+                            setattr(entry_in_session, field, value)
+                        else:
```

Do you need to raise an error? or is a warning alone enough?

---

Actually, I take that back, I think it'd be confusing if some fields updated, and others didn't because of a typo of a field value. I'm going to raise an error instead, so a user can just make whatever edit needed and run the entire function again.